apps/passport-client/components/modals/AccountImportModal.tsx

```diff
@@ -0,0 +1,253 @@
+import { EmailPCDTypeName } from "@pcd/email-pcd";
+import { deserializeStorage } from "@pcd/passport-interface";
+import { Spacer } from "@pcd/passport-ui";
+import { PCDCollection } from "@pcd/pcd-collection";
+import { PCD } from "@pcd/pcd-types";
+import { SemaphoreGroupPCDTypeName } from "@pcd/semaphore-group-pcd";
+import { SemaphoreIdentityPCDTypeName } from "@pcd/semaphore-identity-pcd";
+import { useCallback, useEffect, useState } from "react";
+import styled from "styled-components";
+import { useFilePicker } from "use-file-picker";
+import { useDispatch, useModal, usePCDCollection } from "../../src/appHooks";
+import { getPackages } from "../../src/pcdPackages";
+import { Button } from "../core";
+
+// There are three main UI states that can occur after a user selects a file
+// to import.
+type ImportState =
+  | { state: "ready" }
+  // The selected file is valid, and the user can decide whether to import it.
+  | {
+      state: "valid-file-selected";
+      collection: PCDCollection;
+      pcdsToMergeIds: Set<PCD["id"]>;
+    }
+  // The import has been carried out, and `added` is the number of PCDs
+  // imported.
+  | { state: "import-complete" }
+  // The selected file is not valid.
+  | { state: "invalid-file" };
+
+export function AccountImportModal() {
+  const { openFilePicker, filesContent } = useFilePicker({
+    accept: ".json",
+    multiple: false,
+    onFilesSelected: () => {
+      setImportState({ state: "ready" });
+    }
+  });
+
+  const modal = useModal();
```

Comment:
I am not a huge fan of using a modal for this, and I think this should be a separate screen. One issue with using a modal for this is that currently we only have the capability to have one modal up at a time. This means that the process of importing something can be interrupted irrecoverably by some other piece of code. For example, the sync code runs periodically in the background, and can cause a 'user-invalid' modal to pop up and replace this one. Not the greatest example, since that modal blocks all further interaction anyways, but I think it gets my point across. tagging #1346 here as I think this is a manifestation of the issue.